doc/src/guides/solving/solve-equation-algebraically.md

```diff
@@ -0,0 +1,379 @@
+{-sqrt(y), sqrt(y)}
+>>> for solution_set in solutions_set:
+...     print(solution_set)
+sqrt(y)
+-sqrt(y)
+```
+    
+### Parse a string representing the equation
+
+Parse a string representing the equation into a form that SymPy can understand (`Eq` form), 
+then apply {func}`~.solve` to the parsed expression.  
+This approach is convenient if you are programmatically reading in a string. 
+We [recommend against using parsing a string if you are creating the expression yourself](https://github.com/sympy/sympy/wiki/Idioms-and-Antipatterns#strings-as-input). 
+Parsing an equation from a string requires you to use 
+{func}`transformations <sympy.parsing.sympy_parser.parse_expr>`
+for SymPy to interpret equals signs and create symbols from your variables.
+
+You should always include the variable to solve for if you want to extract results programmatically, 
+to ensure that SymPy solves for the desired variable. 
+To ensure SymPy will produce results in a consistent format, use `dict=True`. 
+To extract the solutions, you can iterate through the list of dictionaries:  
+    
+```py
+>>> from sympy import parse_expr, solve, solveset
+>>> from sympy.abc import x
+>>> expr = "x ** 2 = y"
+>>> parsed = parse_expr(expr, transformations="all")
+>>> print(parsed)
+Eq(x**2, y)
+>>> solutions = solve(parsed, x, dict=True)
+>>> print(solutions)
+[{x: -sqrt(y)}, {x: sqrt(y)}]
+>>> for solution in solutions:
+...     for key, val in solution.items():
+...         print(val)
+-sqrt(y)
+sqrt(y)
+>>> solutions_set = solveset(parsed, x)
+>>> print(solutions_set)
+{-sqrt(y), sqrt(y)}
+```
+
+If you already have the equation in `Eq` form, you can parse that string:
+
+```py
+>>> from sympy import parse_expr, solve, solveset
+>>> from sympy.abc import x
+>>> expr = "Eq(x**2, y)"
+>>> parsed = parse_expr(expr)
+>>> print(parsed)
+Eq(x**2, y)
+>>> solutions = solve(parsed, x, dict=True)
+>>> print(solutions)
+[{x: -sqrt(y)}, {x: sqrt(y)}]
+>>> for solution in solutions:
+...     for key, val in solution.items():
+...         print(val)
+-sqrt(y)
+sqrt(y)
+>>> solutions_set = solveset(parsed, x)
+>>> print(solutions_set)
+{-sqrt(y), sqrt(y)}
+```
+
+### Restrict the domain of solutions
+
+By default, SymPy will return solutions in the complex domain, which also includes 
+purely real and imaginary values. Here, the first two solutions are real, 
+and the last two are imaginary:
+
+```py
+>>> from sympy import Symbol, solve, solveset
+>>> x = Symbol('x')
+>>> solution = solve(x ** 4 - 256, x, dict=True)
+>>> print(solution)
+[{x: -4}, {x: 4}, {x: -4*I}, {x: 4*I}]
+>>> solution_set = solveset(x ** 4 - 256, x)
+>>> print(solution_set)
+{-4, 4, -4*I, 4*I}
+```
+
+If you want to restrict returned solutions to real numbers, you can
+- for {func}`~.solve`, place an assumption on the symbol to be solved for, $x$
+
+```py
+from sympy import Symbol, solve, solveset
+x = Symbol('x', real=True)
+solution = solve(x ** 4 - 256, x, dict=True)
+print(solution)
+[{x: -4}, {x: 4}]
+```
```

Comment:
You can also include inequalities to limit variables to real values:
```python
>>> solve(((x**2+2)*(x-3)*(x-2)*(x-1),x>0,x<3))
Or(Eq(x, 1), Eq(x, 2))
```
Including only `x>0` will then include `Eq(x, 3)`. But you can't currently inlcude discontinuous ranges since something like `Or(x<0,x>2)` will raise an error.